src/main/scala/workflow/Estimator.scala

```diff
@@ -1,21 +1,40 @@
-package pipelines
+package workflow
 
 import java.io.Serializable
 
 import org.apache.spark.rdd.RDD
 
+import scala.reflect.ClassTag
+
 /**
  * An estimator has a `fit` method which takes an input and emits a [[Transformer]]
  * @tparam A The type of input this estimator (and the resulting Transformer) takes
  * @tparam B The output type of the node this estimator produces when being fit
  */
-abstract class Estimator[A, B] extends Serializable {
+abstract class Estimator[A, B] extends EstimatorNode  {
   /**
-   * An estimator has a `fit` method which emits a pipeline node.
+   * An estimator has a `fit` method which emits a [[Transformer]].
    * @param data Input data.
-   * @return A PipelineNode (Transformer) which can be called on new data.
+   * @return A [[Transformer]] which can be called on new data.
+   */
+  protected def fit(data: RDD[A]): Transformer[A, B]
+
+  private[workflow] final def fit(dependencies: Seq[RDD[_]]): TransformerNode[_] = fit(dependencies.head.asInstanceOf[RDD[A]])
+
+  /**
+   * Constructs a pipeline from a single estimator and training data.
+   * Equivalent to `Pipeline() andThen (estimator, data)`
+   *
+   * @param data The training data
    */
-  def fit(data: RDD[A]): Transformer[A, B]
+  def withData(data: RDD[A]): Pipeline[A, B] = {
+    val nodes: Seq[Node] = Seq(DataNode(data), this, new DelegatingTransformer[B](this.label + ".fit"))
+    val dataDeps = Seq(Seq(), Seq(0), Seq(Pipeline.SOURCE))
+    val fitDeps = Seq(Seq(), Seq(), Seq(1))
+    val sink = 2
```

Comment:
Mmm I don't think this makes sense as a constant in the Pipeline object. It's only two here as an artifact of the nodes Seq being constructed above having the DelegatingTransformer as the last item, it doesn't hold any special meaning beyond this specific graph construction. I could replace this with `nodes.size + 1` if you prefer, or add some comments explaining it?